sourcebundle/builder.go

```diff
@@ -73,12 +73,16 @@ type Builder struct {
 	// selected version of each module registry package.
 	resolvedRegistry map[registryPackageVersion]sourceaddrs.RemoteSource
 
+	// resolvedRegistryVersionDeprecations tracks potential deprecations for
+	// each package version
+	packageVersionDeprecations map[registryPackageVersion]*RegistryVersionDeprecation
```

Comment:
I'm curious why this is a map of pointer-to-`RegistryVersionDeprecation`. Later on we store `nil` in this map if a registry package version has no deprecation information, rather than having no entry in the map at all. Is that important?

Having a pointer here means we need to be diligent about nil checks, or we'll panic. Would there be a disadvantage to this being `map[registryPackageVersion]RegistryVersionDeprecation` instead?

Reply:
This comes from the approach we've taken with how we denote deprecated versions. In essence, when there are no deprecations, the registry api doesn't include it in the Module Version response. So in this way`nil` denotes an un-deprecated module.